network-config-analyzer/K8sPolicyYamlParser.py

```diff
@@ -266,39 +273,45 @@ def parse_policy(self):
         :return: a K8sNetworkPolicy object with proper PeerSets and ConnectionSets
         :rtype: K8sNetworkPolicy
         """
-        if not isinstance(self.policy, dict):
-            self.syntax_error('Top ds is not a map')
+        if not isinstance(self.policy, dict):  # we don't get here
+            self.syntax_error('type of Top ds is not a map')
         if self.policy.get('kind') != 'NetworkPolicy':
             return None  # Not a NetworkPolicy object
-        api_version = self.policy.get('apiVersion', self.policy.get('api_version', ''))
-        if not api_version:
-            self.syntax_error('An object with no specified apiVersion', self.policy)
+        self.check_keys_are_legal(self.policy, 'NetworkPolicy', {'kind': [1, str], 'metadata': [1, dict],
+                                                                 'spec': [0, dict], 'apiVersion': [1, str]},
+                                  {'apiVersion': ['networking.k8s.io/v1', 'extensions/v1beta1']})
+        api_version = self.policy.get('apiVersion')
         if 'k8s' not in api_version and api_version != 'extensions/v1beta1':
             return None  # apiVersion is not properly set
-        if api_version not in ['networking.k8s.io/v1', 'extensions/v1beta1']:
-            raise Exception('Unsupported apiVersion: ' + api_version)
-        self.check_keys_are_legal(self.policy, 'networkPolicy', {'kind': 1, 'metadata': 1, 'spec': 1,
-                                                                 'apiVersion': 0, 'api_version': 0})
-        if 'name' not in self.policy['metadata']:
-            self.syntax_error('NetworkPolicy has no name', self.policy)
-        if 'namespace' in self.policy['metadata']:
-            self.namespace = self.peer_container.get_namespace(self.policy['metadata']['namespace'])
-        res_policy = K8sNetworkPolicy(self.policy['metadata']['name'], self.namespace)
+
+        policy_metadata = self.policy['metadata']
+        allowed_metadata_keys = {'name': [1, str], 'namespace': [0, str], 'annotations': 0, 'clusterName': 0,
+                                 'creationTimestamp': 0, 'deletionGracePeriodSeconds': 0, 'deletionTimestamp': 0,
+                                 'finalizers': 0, 'generateName': 0, 'generation': 0, 'labels': 0, 'managedFields': 0,
+                                 'ownerReferences': 0, 'resourceVersion': 0, 'selfLink': 0, 'uid': 0}
+        self.check_keys_are_legal(policy_metadata, 'metadata', allowed_metadata_keys)
+        pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+        if re.fullmatch(pattern, policy_metadata['name']) is None:
```

Comment:
I'm not sure I fully understand this. Let's discuss

Reply:
Copying an error message that I got when running tests on live k8s cluster (in my test : name included **_** ):

> The NetworkPolicy "metadata_name_test7" is invalid: metadata.name: Invalid value: "metadata_name_test7": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

after consulting with Ziv , we want to follow k8s behavior here too, so I used the same k8s regex to our checks.